plugins/inputs/proxysql/proxysql.go

```diff
@@ -0,0 +1,552 @@
+package proxysql
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+type Proxysql struct {
+	Servers                []string `toml:"servers"`
+	GatherGlobal           bool     `toml:"gather_global"`
+	GatherCommandsCounters bool     `toml:"gather_commands_counters"`
+	GatherConnectionPool   bool     `toml:"gather_connection_pool"`
+	GatherUsers            bool     `toml:"gather_users"`
+	GatherRules            bool     `toml:"gather_rules"`
+	GatherQueries          bool     `toml:"gather_queries"`
+	GatherMemoryMetrics    bool     `toml:"gather_memory_metrics"`
+	GatherProcessList      bool     `toml:"gather_process_list"`
+	IntervalSlow           string   `toml:"interval_slow"`
+	tls.ClientConfig
+}
+
+var sampleConfig = `
+  ## specify servers via a url matching:
+  ##  [username[:password]@][protocol[(address)]]/[?tls=[true|false|skip-verify|custom]]
+  ##  see https://github.com/go-sql-driver/mysql#dsn-data-source-name
+  ##  e.g.
+  ##    servers = ["user:passwd@tcp(127.0.0.1:6032)/?tls=false"]
+  ##    servers = ["user@tcp(127.0.0.1:6032)/?tls=false"]
+  #
+  ## If no servers are specified, then localhost is used as the host.
+  servers = ["tcp(127.0.0.1:6032)/"]
+  #
+  ## gather metrics from stats_mysql_global
+  gather_global                             = true  
+  ## gather metrics from stats_mysql_commands_counters
+  gather_commands_counters                  = true
+  ## gather metrics from stats_mysql_connection_pool
+  gather_connection_pool                    = true
+  ## gather metrics from stats_mysql_users
+  gather_users                              = true
+  ## gather metrics from stats_mysql_query_rules
+  gather_rules                              = true
+  ## gather metrics from stats_mysql_query_digest
+  gather_queries                            = true
+  ## gather metrics from stats_memory_metrics
+  gather_memory_metrics                     = true
+  ## gather thread state counts from stats_mysql_processlist
+  gather_process_list                       = true
+  #
+  ## Some queries we may want to run less often (such as SHOW GLOBAL VARIABLES)
+  interval_slow                             = "30m"
```

Comment:
Is this option needed on proxysql?  I would like to improve the way this works, perhaps design a pattern for this type of functionality, but don't have the time to work on it currently and if we can leave it out then it would keep our options more open.  If proxysql has as many global variables as mysql it could be required.